src/Pages/Home.js

```diff
@@ -1,46 +1,207 @@
-  baseKey: process.env.REACT_APP_AIRTABLE_BASE_KEY,
-};
-const base = new Airtable({ apiKey: airtableConfig.apiKey })
-  .base(airtableConfig.baseKey);
-
-function Home() {
-  const [collections, setCollections] = useState([]);
-  const getCollections = () => {
-    base('Collection').select({ view: 'Grid view' }).all() // Gets + returns all records
-      .then((records) => { // Takes in returned records + calls setPosts to store in posts arr
-        setCollections(records);
-      });
-  };
-
-  useEffect(getCollections, []);
-  return (
-    <div>
-      <h1 className="headings">Book Recommendation Hub</h1>
-      <h2 className="headings">Collections</h2>
-      <CollectionsCarousel
-        elementArray={collections}
-        slidesAtATime={6}
-        prevArrow={LeftArrow}
-        nextArrow={RightArrow}
-        widthPercent={100}
-        spaceBetweenEntries={16}
-        swiperHeight={120}
-        cardImageHeightPercent={80}
-        cardImageWidthPercent={80}
-      />
-      <h2 className="headings">Books</h2>
-    </div>
-  );
-}
-
-export default Home;
+import React, { useState, useEffect } from 'react';
+import { NavLink } from 'react-router-dom';
+import Button from '@mui/material/Button';
+import Stack from '@mui/material/Stack';
+import CollectionsCarousel from '../Components/CollectionsComponents/CollectionsCarousel';
+import Carousel from '../Components/CreatorPage/BookCarousel';
+import LeftArrow from '../Assets/Images/left-arrow.svg';
+import RightArrow from '../Assets/Images/right-arrow.svg';
+import './PagesTemp.css';
+
+// Airtable Configuration
+const Airtable = require('airtable');
+
+const airtableConfig = {
+  apiKey: process.env.REACT_APP_AIRTABLE_USER_KEY,
+  baseKey: process.env.REACT_APP_AIRTABLE_BASE_KEY,
+};
+const base = new Airtable({ apiKey: airtableConfig.apiKey })
+  .base(airtableConfig.baseKey);
+
+const styles = {
+  buttonStack: {
+    backgroundColor: '#EAF3FE',
+    color: '#1A296A',
+    padding: '10px',
+    width: '12vw',
+    borderRadius: '10px',
+    fontfamily: 'Work Sans',
+    fontSize: '15px',
+    textTransform: 'none',
+    fontWeight: 'bold',
+    fontStyle: 'normal',
+  },
+};
+
+// main function
+function Home() {
+  const [collections, setCollections] = useState([]);
+  const [newReleases, setNewReleases] = useState([]);
+
+  // Collections Carousel part
+  const getCollections = () => {
+    base('Collection').select({ view: 'Grid view' }).all() // Gets + returns all records
+      .then((records) => { // Takes in returned records + calls setPosts to store in posts arr
+        setCollections(records);
+      });
+  };
+
+  // New Releases filter function
+  // filters for the first 14 books sorted by latest publishing date
+  const NewReleasesFunction = () => new Promise((resolve, reject) => {
+    base('Book').select({
+      sort: [{ field: 'date_added', direction: 'desc' }],
+      maxRecords: 14,
+      view: 'Grid view',
+    }).all().then((records) => {

```

Comment:
i have spent the last 50 mins looking for docs for .all() and there is nothing out there on the web :'( but if .all ensures that all the records in the .then() are fulfilled, then you might be able to just do setNewReleases inside this function instead of returning a promise! this implementation will still work tho

Reply:
Yeah I agree! I think Ryan might have been prepping to do smth specific requiring iterating over the objects, but I don't think it's necessary as is.

Reply:
i think this might have been inspired by one of my code snippets in BookCardsDisplay that i realize has the same issue, oops